### PR TITLE
[DOCFIX] Add docs for rolling upgrade and decommission worker

### PR DESCRIPTION
**Pull request by jiacheliu3**
### What changes are proposed in this pull request?

This is the doc update from a few recent commits:
1. https://github.com/Alluxio/alluxio/commit/aee3c5cb960b2e472f4fcdb86e731ca3b91d2f8c
2. https://github.com/Alluxio/alluxio/commit/9a4e154e7293667572fec25c63136462fb58a345
3. https://github.com/Alluxio/alluxio/commit/dbac084c1e1d89c1e9253da036746e2df9e1ef14
4. https://github.com/Alluxio/alluxio/commit/8da59539201cff376c33256846bd565d976a148e
5. https://github.com/Alluxio/alluxio/commit/141ee0e567a4c6c60907f85ea66b828704bd761d
6. https://github.com/Alluxio/alluxio/commit/26257e6f35c57d937f6f5bdee831facc2fcf7e1a

In short, a few improvements have been added that the admin can now see the version/revision of the running Alluxio components in the cluster. A few commands were also added to help with gracefully decommissioning/restarting workers in the cluster.

### Why are the changes needed?

This doc change helps users understand how to utilize those features.

### CODE REVIEW

File: docs/en/operation/Admin-CLI.md [hunk 1]
```diff
@@ -301,9 +319,15 @@ hdfs://localhost:9000/ on / (hdfs, capacity=-1B, used=-1B, not read-only, not sh
 
 ```shell
 $ bin/alluxio fsadmin report jobservice
-Worker: MigrationTest-workers-2  Task Pool Size: 10     Unfinished Tasks: 1303   Active Tasks: 10     Load Avg: 1.08, 0.64, 0.27
```
Comment (jiacheliu3) [inline docs/en/operation/Admin-CLI.md:304]: the format has been changed to the table-flavor format below

File: docs/en/operation/Admin-CLI.md [hunk 2]
```diff
@@ -362,3 +396,158 @@ alluxio.master.audit.logging.enabled
...
+This command returns either when all workers are idle, or when this wait time is up. The default value is `5m`.
+
+`--disable/-d` is an optional argument. If this is specified, the decommissioned workers will not be able to
+register to the master. In other words, a disabled worker cannot join the cluster and will not be chosen for I/O requests.
+This is useful when the admin wants to remove the workers from the cluster. When those disabled workers register,
+the master will reject them but will not kill the worker processes. This is often used in pair with the `enableWorker` command.
+
+The command will perform the following actions:
+
+1. For each worker in the batch, send a decommission command to the primary Alluxio master so
```
Comment (dbw9580) [inline docs/en/operation/Admin-CLI.md:429]: Do you think this deserves a dedicated section in [`Administration > Upgrade`](https://docs.alluxio.io/ee/user/stable/en/administration/Upgrade.html)? This `Admin CLI` is mostly focusing on the command and its parameters, it would be better to present the overall process of cluster rolling upgrade in a more mission-centric way.

File: docs/en/operation/Admin-CLI.md [hunk 3]
```diff
@@ -362,3 +396,158 @@ alluxio.master.audit.logging.enabled
...
+This command can be typically used for the following use cases:
+1. Perform a graceful rolling restart of all workers in the cluster, where no user requests should fail.
+2. Scale down the cluster without interrupting user I/O workflow.
+
+```shell
+$ ./bin/alluxio fsadmin decommissionWorker --addresses data-worker-0,data-worker-1 [--wait 5m] [--disable]
+```
+The arguments are explained as follows:
+
+`--address/-a` is a required argument, followed by a list of comma-separated worker addresses. Each worker address is `<host>:<web port>`.
```
Comment (dbw9580) [inline docs/en/operation/Admin-CLI.md:414]: is it `--address` or `--addresses`?

File: docs/en/operation/Admin-CLI.md [hunk 4]
```diff
@@ -362,3 +396,158 @@ alluxio.master.audit.logging.enabled
...
+Unlike many other commands which specify the RPC port, we use the web port here because the command will monitor the worker's workload
+exposed at the web port. If the port is not specified, the value in `alluxio.worker.web.port` will be used. Note that `alluxio.worker.web.port`
+will be resolved from the node where this command is run.
+
+`--wait/-w` is an optional argument. This argument defines how long this command waits for the workers to become idle.
```
Comment (dbw9580) [inline docs/en/operation/Admin-CLI.md:419]: How is a worker being "idle" defined? When a worker is deemed idle, what promises concerning data integrity and operational stability can the user expect and _not_ expect?

File: docs/en/operation/Admin-CLI.md [hunk 5]
```diff
@@ -362,3 +396,158 @@ alluxio.master.audit.logging.enabled
...
+
+**Limitations**
+
+In some cases, the `decommissionWorker` command may return code 0 (success) but when a worker process is killed,
+some user I/O may fail.
+
+When the `decommissionWorker` command waits for a worker to become idle, it only respects ongoing I/O requests on the worker.
+If the Alluxio client is reading/writing the worker with short circuit, the client directly reads/writes cache files
+in worker storage and maintains a gRPC stream with the worker simply for locking that block. An open(and idle) stream
+will NOT be respected by the `decommissionWorker` command so it may consider the worker is idle.
```
Comment (dbw9580) [inline docs/en/operation/Admin-CLI.md:448]: This can be used to define worker idleness.

File: docs/en/operation/Admin-CLI.md [hunk 6]
```diff
@@ -362,3 +396,158 @@ alluxio.master.audit.logging.enabled
...
+However, when the admin kills the worker and deletes the cache storage, the client request will fail
+either because the cache blocks are gone or because the gRPC stream is broken. So if you are using short circuit,
+wait a few more minutes before killing the worker process and deleting cached blocks. The clients will
+know the workers are decommissioned (should not be read/written) and stop using those workers by short circuit.
+
+The `decommissionWorker` command does NOT consider cache blocks on the target workers. That means if
+decommissioning some workers will bring ALL replicas of certain blocks offline, and those blocks only exist in cache,
```
Comment (dbw9580) [inline docs/en/operation/Admin-CLI.md:455]:
```suggestion
there is a block whole sole replica happens to reside on one of the workers that will be decommissioned,
and it's not in UFS either (due to `MUST_CACHE` or `ASYNC_THROUGH` write mode,
```

File: docs/en/operation/Admin-CLI.md [hunk 7]
```diff
@@ -362,3 +396,158 @@ alluxio.master.audit.logging.enabled
...
+
+1. `0(OK)`: All workers are successfully decommissioned and now idle. Safe to kill or restart this batch of workers now.
+2. `1(DECOMMISSION_FAILED)`: Failed to decommission all workers. The admin should double check the worker addresses and the primary master status.
+3. `2(LOST_MASTER_CONNECTION)`: Lost connection to the primary master while this command is running. This suggests the configured master address is wrong or the primary master failed over.
+4. `3(WORKERS_NOT_IDLE)`: Some workers were still not idle after the wait. Either the wait time is too short or those workers failed to mark decommissioned. The admin should manually intervene and check those workers.
+5. `10(LOST_SOME_WORKERS)`: Workers are decommissioned but some or all workers lost contact while this command is running. If a worker is not serving then it is safe to kill or restart. But the admin is advised to double check the status of those workers.
+
+### enableWorker
+
+The `enableWorker` command is the reverse operation of `decommissionWorker -d`.
```
Comment (dbw9580) [inline docs/en/operation/Admin-CLI.md:539]: Explain in what circumstances such a command would be useful?

**Review by dbw9580 — CHANGES_REQUESTED** [review]

File: docs/en/operation/Admin-CLI.md (hunk 3 above)
Comment (jiacheliu3) [reply docs/en/operation/Admin-CLI.md:414]: tks

File: docs/en/operation/Admin-CLI.md (hunk 6 above)
Comment (jiacheliu3) [reply docs/en/operation/Admin-CLI.md:457]: It can also be the case that one block has multiple replicas, and all replicas are on this batch of workers.

File: docs/en/operation/Admin-CLI.md (hunk 2 above)
Comment (jiacheliu3) [reply docs/en/operation/Admin-CLI.md:429]: added a few sections to the `Upgrade` page, PTAL

File: docs/en/operation/Admin-CLI.md (hunk 4 above)
Comment (jiacheliu3) [reply docs/en/operation/Admin-CLI.md:419]: added a paragraph for correctness

File: docs/en/operation/Admin-CLI.md (hunk 7 above)
Comment (jiacheliu3) [reply docs/en/operation/Admin-CLI.md:474]: added a few lines

**Review by dbw9580 — APPROVED** [review]
LGTM

**Comment by jiacheliu3** [comment]
alluxio-bot, cherry-pick this to branch-2.10 please

File: docs/en/administration/Upgrade.md [hunk 8]
```diff
@@ -87,6 +87,150 @@ $ ./bin/alluxio-start.sh all
 ```
 5. If you have updated the Alluxio client jar for an application, restart that application to use the new Alluxio client jar.
 
+### Rolling upgrade/restart masters
```
Comment (Xenorith) [inline docs/en/administration/Upgrade.md:90]: the rolling upgrade feature should be considered as experimental. i'll make a new grouping and tag it
